Sources/TuistDependencies/SwiftPackageManager/SwiftPackageManagerInteractor.swift

```diff
@@ -135,7 +135,9 @@ public final class SwiftPackageManagerInteractor: SwiftPackageManagerInteracting
         dependencies: TuistGraph.SwiftPackageManagerDependencies,
         swiftToolsVersion: TSCUtility.Version?
     ) throws {
-        let isLegacy = TSCUtility.Version(string: try System.shared.swiftVersion())! < TSCUtility.Version(5, 6, 0)
+        // Use ProjectDescription.Version to correctly parse version when patch or minor is missing (for example, "5.6")
+        let version = ProjectDescription.Version(string: try System.shared.swiftVersion())!
+        let isLegacy = TSCUtility.Version(version.major, version.minor, version.patch) < TSCUtility.Version(5, 6, 0)
```

Comment:
If we update swift-tools-support-core we can just use

```suggestion
        let isLegacy = try TSCUtility.Version(versionString: try System.shared.swiftVersion(), usesLenientParsing: true) < TSCUtility.Version(5, 6, 0)
```